ksqldb-engine/src/main/java/io/confluent/ksql/util/PersistentQueryMetadataImpl.java

```diff
@@ -226,4 +228,32 @@ public synchronized void stop() {
   public Optional<ScalablePushRegistry> getScalablePushRegistry() {
     return scalablePushRegistry;
   }
+
+  private static final class ScalablePushQueryListener implements Listener {
```

Comment:
This technically also calls the other listener, so maybe give it a more generic name like PushQueryListenerWrapper?